bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py

```diff
@@ -0,0 +1,200 @@
+from invenio.refextract_linker import find_doi, find_journal, find_reportnumber, find_book, find_isbn
+from invenio.dbquery import run_sql
+
+def get_institution_ids(text, INSTITUTION_CACHE={}):
+    # HACK: I know... I am sorry for that. It's for a good cause
+    # FIXME: use redis
+    if text not in INSTITUTION_CACHE:
+        INSTITUTION_CACHE[text] = perform_request_search(cc='Institutions', p='institution:"%s"' % text)
+    return INSTITUTION_CACHE[text]
+
+def reference2citation_element(subfields):
+    citation_element = {}
+    for code, value in subfields:
+        if code == 'a':
+            citation_element['doi_string'] = value
+        elif code == 't':
+            citation_element['title'] = value
+        elif code == 'i':
+            citation_element['isbn'] = value
+        elif code == 'r':
+            citation_element['report_num'] = value
+        elif code == 'y':
+            citation_element['year'] = value
+        elif code == 's':
+            try:
+                journal, volume, page = value.split(',')
+                citation_element['journal_title'] = journal
+                citation_element['volume'] = volume
+                citation_element['page'] = page
+            except ValueError:
+                pass
+    if 'journal_title' in citation_element:
+        # HACK: to work around clash between find_journal() and find_book()
+        citation_element['title'] = citation_element['journal_title']
+    return citation_element
+
+
+def get_matched_id(subfields):
+    citation_element = reference2citation_element(subfields)
+    if 'doi_string' in citation_element:
+        recids = find_doi(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    if 'journal_title' in citation_element and 'year' in citation_element:
+        recids = find_journal(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    if 'report_num' in citation_element:
+        recids = find_reportnumber(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    if 'isbn' in citation_element:
+        recids = find_isbn(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    #if 'title' in citation_element:
+        #recids = find_book(citation_element)
+        #if len(recids) == 1:
+            #return recids.pop()
+    return None
+
+
+def format_element(bfo, oai=0):
+    """Produce MARCXML with enhanced fields.
+
+    Adds 100/700 $x with Record ID of linked HepName,
+                 $y with True/False if the signature is claimed
+                 $z with Record ID of institution
+         371/110 $z with Record ID of institution
+         502     $z with Record ID of institution
+         999C5   $0 with on the fly discovered Record IDs (not for books)
+         773     $0 with Record ID of corresponding Journal or Conference
+         693/710 $0 with Record ID of corresponding experiment
+    """
```

Comment:
@annetteholtkamp, @tsgit, @ksachs what do you think? Do you know if there are other hidden links that can be made more explicit?